src/error_chain.rs

```diff
@@ -137,6 +137,17 @@ macro_rules! error_chain_processed {
                 )
             }
 
+            /// Construct a chained error from another boxed error and a kind, and generates a backtrace
+            pub fn with_boxed_chain<K>(error: Box<::std::error::Error + Send>, kind: K)
+                -> $error_name
+                where K: Into<$error_kind_name>
+            {
+                $error_name(
+                    kind.into(),
+                    $crate::State::new::<$error_name>(error, ),
+                )
+            }
+
```

Comment:
Well, I guess that's me nitpicking about having two things in the same commit XD
Box being a strange builtin and specialization being missing can both be a pain when dealing with generic implementation, I had this kind of problem in error-chain... I guess we can't do anything about it for now. Could you just rewrite with_chain so that it uses with_boxed_chain internally? Also please update the changelog.